### ISSUE TITLE
Compiling Clang in Windows get issues with min and max

### ISSUE THREAD

**Issue by Togtja** [kcat/openal-soft#566]
So I seem to be suffering from the issue outline in this stack overflow [question ](https://stackoverflow.com/questions/11544073/how-do-i-deal-with-the-max-macro-in-windows-h-colliding-with-max-in-std)

Where ever you use `std::numeric_limits<int>::max()` or `std::max()` the Windows API believes you to be using it's marco for `min/max` the solution is to define `NOMINMAX` or change all the `std` to have parentheses around it like this: `(std::numeric_limits<int>::max)()`

**Comment by kcat** [comment]
Where is the error happening? `NOMINMAX` s defined here:
https://github.com/kcat/openal-soft/blob/1d7ff54f/CMakeLists.txt#L208
and applied to the library here:
https://github.com/kcat/openal-soft/blob/1d7ff54f/CMakeLists.txt#L1369

**Comment by Togtja** [comment]
Alot of places. So I use the newest Clang (12.0.0) compiling openal-soft's master. I first get stopped on:
```
[build] ../../../common/threads.cpp:67:74: error: too few arguments provided to function-like macro invocation
[build]     if(initial > static_cast<unsigned int>(std::numeric_limits<int>::max()))
....
[build] ../../../common/threads.cpp:69:76: error: too few arguments provided to function-like macro invocation
[build]     mSem = CreateSemaphore(nullptr, initial, std::numeric_limits<int>::max(), nullptr);
````
Then after solving those. I get stopped on:
```
[build] ../../../common/almalloc.h:91:54: error: too few arguments provided to function-like macro invocation
[build]         if(n > std::numeric_limits<std::size_t>::max()/sizeof(T)) throw std::bad_alloc();
```
Then I get stopped:
```
[build] ../../../alc/backends/wasapi.cpp:129:76: error: too few arguments provided to function-like macro invocation
[build]     return static_cast<uint>(mini64(retval, std::numeric_limits<uint>::max()));
...
[build] ../../..\al/source.h:49:55: error: too few arguments provided to function-like macro invocation
[build]     float MaxDistance{std::numeric_limits<float>::max()};
...
[build] ../../../alc/alc.cpp:2665:47: error: too few arguments provided to function-like macro invocation
[build]                 std::numeric_limits<int>::max()));
```
And it continues like this, if you want the rest, please let me know, and I can go through them

**Comment by Togtja** [comment]
My gut feeling is that if`(MSVC)` does not pick up on me using Clang on windows

**Comment by Togtja** [comment]
So I added this to the cmake:
`message(STATUS "What is ${MSVC} and we should probably have ${WIN32}")`
and it returned
`What is  and we should probably have 1`
So I think my gut feeling was correct

**Comment by Togtja** [comment]
A solution that worked for me was to add:
```
if(WIN32 AND NOT MSVC)
    set(CPP_DEFS ${CPP_DEFS} _CRT_SECURE_NO_WARNINGS NOMINMAX)
endif()
```
Though I don't know how this would effect MINGW and the like

**Comment by kcat** [comment]
Fixed with commit 602b33fede0ce7d158ca96cda6db45742b2b2fce.
